lib/path.c

```diff
@@ -1028,7 +1028,12 @@ static int ul_path_cpuparse(struct path_cxt *pc, cpu_set_t **set, int maxcpus, i
 	if (!f)
 		return -errno;
 
-	rc = fgets(buf, len, f) == NULL ? -errno : 0;
+	if (fgets(buf, len, f) == NULL) {
+		if (errno > 0)
```

Comment:
When should this ever be true?

Reply:
The real state would probably be in `feof()` or `ferror()`

Reply:
But I think nothing every checks the return values anyways.

Reply:
> When should this ever be true?

Never, brain fart.

> But I think nothing every checks the return values anyways.

Seems so, but since the set is not usable, it should be ok.

Reply:
Personally I preferred the first version that just returned `-EIO` best.
It is concise and correct.

Whenever something breaks, the specific errorcode is lost anyways further up the callstack so a strace is required anyways.

Reply:
> Personally I preferred the first version that just returned -EIO best.

Fine with me, I've reverted to that version.